File: frontend/src/components/System/InvestmentState/InvestmentState.js
import React from 'react'
import './InvestmentState.css'

const InvestmentState = (props) => {
    switch (props.state) {
        case 'evergreen':
            return <span className="badge investEvergreen">EVERGREEN</span>
        case 'invest':
            return <span className="badge invest">INVEST</span>
        case 'maintain':
            return <span className="badge investMaintain">MAINTAIN</span>
        case 'sunset':
            return <span className="badge investSunset">SUNSET</span>
        case 'decommissioned':
            return <span className="badge investDecommissioned">DECOMMISSIONED</span>
        case 'cancelled':
            return <span className="badge investCancelled">CANCELLED</span>
        default:
            return <span className="badge investmentUnknown">UNKNOWN</span>
    }
}

export default InvestmentState

File: backend/src/main/java/uk/gov/digital/ho/systemregister/application/messaging/commands/validation/InvestmentState.java
package uk.gov.digital.ho.systemregister.application.messaging.commands.validation;

import javax.validation.Constraint;
import javax.validation.Payload;
import javax.validation.constraints.Pattern;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.*;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

//TODO update to custom validator with dynamic values in default message
@Constraint(validatedBy = {})
@Pattern(regexp = "evergreen|invest|maintain|sunset|decommissioned|cancelled|unknown", message = "Investment state must be one of the following values: evergreen, invest, maintain, sunset, decommissioned, cancelled, or unknown")
@Retention(RUNTIME)
@Target({METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER, TYPE_USE})
@SuppressWarnings("unused")
public @interface InvestmentState {
    String message() default "";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}
